django/db/backends/sqlite3/introspection.py

```diff
@@ -65,6 +65,32 @@ def get_field_type(self, data_type, description):
             return 'AutoField'
         return field_type
 
+    def only_nonempty_tables(self, tables):
+        """
+        Take a list of tables, return a list of tables that have at least
+        one row.
+        """
+        if not tables:
+            return []
+        sql_list = []
+        BATCH_SIZE = 999 // 2
```

Comment:
imo we should be batching in the base class too, because there's sure to be *some* limit in other backends too, even if we haven't discovered it/hit it.

Reply:
Would like another opinion

Reply:
Can you suggest someone who can give their input?

Reply:
@felixxm used to work a lot with Oracle , I think he should be able to input about risk of limits?

Reply:
@felixxm able to give your input?

Reply:
I don't see any parameter in [the query](https://github.com/django/django/pull/12482/files#diff-6ab0246e569ff27b731f91a601e3273b963c9629811a6345444719a3e024e278R86), so doing this in batches should not be necessary.

`SQLITE_LIMIT_VARIABLE_NUMBER` in SQLite is the maximum number of parameters, there is a similar restriction in Oracle but with `2^16 - 1` parameters. You can check `features.max_query_params` for each backend. IMO it's not necessary in this solution.

Reply:
SQLite needs batching because it gives: `django.db.utils.OperationalError: too many terms in compound SELECT` otherwise.

Reply:
That's a different restriction related with number of terms, see [`SQLITE_MAX_COMPOUND_SELECT`](https://sqlite.org/limits.html#max_compound_select).